src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -101,6 +104,18 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
             }
         }
 
+        private async Task DocumentChangedAsync(Document oldDocument, Document newDocument)
+        {
+            var oldTextVersion = await oldDocument.GetTextVersionAsync(CancellationToken.None).ConfigureAwait(false);
+            var newTextVersion = await newDocument.GetTextVersionAsync(CancellationToken.None).ConfigureAwait(false);
+
+            if (oldTextVersion != newTextVersion)
+            {
+                UpdateClassVersion();
```

Comment:
With this, each thing is written atomically, but the UI thread could still read at a time where the TypeVersion is updated but the member version isn't.  Is that okay?